ami/models/i_jepa.py

```diff
@@ -0,0 +1,461 @@
+                Specify hidden dims for transformer layers.
+                Defaults to 4.0.
+            qkv_bias (bool):
+                Whether to use bias in MLPs used to get qkv in attention module.
+                Defaults to True.
+            qk_scale (float | None):
+                The multiplier to be applied to the matrix product of q and v in attention module.
+                Defaults to None.
+            drop_rate (float):
+                Ratio of Dropout to be performed within last MLP in each transformer layers.
+                Defaults to 0.0.
+            attn_drop_rate (float):
+                Ratio of Dropout to be performed within MLP for the matrix product of q and k in attention module.
+                Defaults to 0.0.
+            drop_path_rate (float):
+                Maximum value of stochastic depth decay rule.
+                Defaults to 0.0.
+            init_std (float):
+                Std for initializing layers.
+                Defaults to 0.02.
+        """
+        super().__init__()
+        self.num_features = self.embed_dim = embed_dim
+        self.num_heads = num_heads
+        # define input layer to convert input image into patches.
+        self.patch_embed = PatchEmbed(
+            patch_size=patch_size,
+            in_channels=in_channels,
+            embed_dim=embed_dim,
+        )
+        # define positional encodings
+        assert img_size % patch_size == 0
+        n_patches = (img_size // patch_size) ** 2
+        self.positional_encodings: torch.Tensor
+        positional_encodings = get_2d_positional_embeddings(
+            embed_dim,
+            img_size // patch_size,
+        ).reshape(1, n_patches, embed_dim)
+        self.register_buffer("positional_encodings", torch.from_numpy(positional_encodings).float())
+        # define transformers
+        dpr = np.linspace(0, drop_path_rate, depth).tolist()  # stochastic depth decay rule
+        self.vit_layers = nn.ModuleList(
+            [
+                VisionTransformerLayer(
+                    embedding_dim=embed_dim,
+                    num_heads=num_heads,
+                    mlp_ratio=mlp_ratio,
+                    qkv_bias=qkv_bias,
+                    qk_scale=qk_scale,
+                    dropout=drop_rate,
+                    attn_drop=attn_drop_rate,
+                    drop_path=dpr[i],
+                )
+                for i in range(depth)
+            ]
+        )
+        self.norm = nn.LayerNorm(embed_dim, eps=1e-6)
+        # initialize
+        self.init_std = init_std
+        self.apply(self._init_weights)
+        self.fix_init_weight()
+
+    def fix_init_weight(self) -> None:
+        def rescale(param: torch.Tensor, layer_id: int) -> None:
+            param.div_(math.sqrt(2.0 * layer_id))
+
+        for layer_id, layer in enumerate(self.vit_layers, start=1):
+            rescale(layer.attn.proj.weight.data, layer_id)
+            rescale(layer.mlp.fc2.weight.data, layer_id)
```

Comment:
ここもモジュール関数に展開していいと思う。
```suggestion
def fix_init_weight(vit_layers: Sequence[VisionTransformerLayer]) -> None:
        def rescale(param: torch.Tensor, layer_id: int) -> None:
            param.div_(math.sqrt(2.0 * layer_id))

        for layer_id, layer in enumerate(vit_layers, start=1):
            rescale(layer.attn.proj.weight.data, layer_id)
            rescale(layer.mlp.fc2.weight.data, layer_id)
```